server/chat-plugins/scavengers.ts

```diff
@@ -1468,88 +1731,104 @@ const commands = {
 
 		if (!this.can('declare', null, room)) return false; // perms for editing
 
-		let [type, blitzPoints] = target.split(',');
-		blitzPoints = parseInt(blitzPoints);
-		type = toID(type);
-		if (!RATED_TYPES.includes(type)) return this.errorReply(`You cannot set blitz points for ${type} hunts.`);
+		const parts = target.split(',');
+		const blitzPoints = parseInt(parts[1]);
+		const gameType = toID(parts[0]) as GameTypes;
+		if (!RATED_TYPES.includes(gameType)) return this.errorReply(`You cannot set blitz points for ${gameType} hunts.`);
 
-		if (isNaN(blitzPoints) || blitzPoints < 0 || blitzPoints > 1000) return this.errorReply("The points value awarded for blitz must be an integer bewteen 0 and 1000.");
-		if (!room.blitzPoints) room.blitzPoints = {};
-		room.blitzPoints[type] = blitzPoints;
+		if (isNaN(blitzPoints) || blitzPoints < 0 || blitzPoints > 1000) {
+			return this.errorReply("The points value awarded for blitz must be an integer bewteen 0 and 1000.");
+		}
+		if (!room.scavSettings.blitzPoints) room.scavSettings.blitzPoints = {};
+		room.scavSettings.blitzPoints[gameType] = blitzPoints;
 
 		if (room.chatRoomData) {
-			room.chatRoomData.blitzPoints = room.blitzPoints;
+			room.chatRoomData.scavSettings = room.scavSettings;
 			Rooms.global.writeChatRoomData();
 		}
-		this.privateModAction(`(${user.name} has set the points awarded for blitz for ${type} hunts to ${blitzPoints}.)`);
-		this.modlog('SCAV BLITZ', null, `${type}: ${blitzPoints}`);
+		this.privateModAction(`(${user.name} has set the points awarded for blitz for ${gameType} hunts to ${blitzPoints}.)`);
+		this.modlog('SCAV BLITZ', null, `${gameType}: ${blitzPoints}`);
 
 		// double modnote in scavs room if it is a subroomgroupchat
-		if (room.parent && !room.chatRoomData) {
-			scavsRoom.modlog(`(scavengers) SCAV BLITZ: by ${user.id}: ${type}: ${blitzPoints}`);
-			scavsRoom.sendMods(`(${user.name} has set the points awarded for blitz for ${type} hunts to ${blitzPoints} in <<${room.roomid}>>.)`);
-			scavsRoom.roomlog(`(${user.name} has set the points awarded for blitz for ${type} hunts to ${blitzPoints} in <<${room.roomid}>>.)`);
+		if (room.parent && !room.chatRoomData && scavsRoom) {
+			scavsRoom.modlog(`(scavengers) SCAV BLITZ: by ${user.id}: ${gameType}: ${blitzPoints}`);
+			scavsRoom.sendMods(`(${user.name} has set the points awarded for blitz for ${gameType} hunts to ${blitzPoints} in <<${room.roomid}>>.)`);
+			scavsRoom.roomlog(`(${user.name} has set the points awarded for blitz for ${gameType} hunts to ${blitzPoints} in <<${room.roomid}>>.)`);
 		}
 	},
 
 	sethostpoints(target, room, user) {
-		if (room.roomid !== 'scavengers' && !(room.parent && room.parent.roomid === 'scavengers')) return this.errorReply("This command can only be used in the scavengers room.");
+		if (room.roomid !== 'scavengers' && !(room.parent && room.parent.roomid === 'scavengers')) {
+			return this.errorReply("This command can only be used in the scavengers room.");
+		}
```

Comment:
Maybe make it like this:

```ts
function getScavsRoom(room?: Room | null) {
	if (!room) return Rooms.getRoom('scavengers');
	if (room.roomid === 'scavengers') return room;
	if (room.parent?.roomid === 'scavengers') return room.parent;
	return null;
}
```

And then this would be:

```suggestion
		const scavsRoom = getScavsRoom(room);
		if (!scavsRoom) {
			return this.errorReply("This command can only be used in the scavengers room.");
		}
```